src/InfoBoxes/InfoBoxWindow.cpp

```diff
@@ -346,6 +347,10 @@ bool
 InfoBoxWindow::OnKeyDown(unsigned key_code)
 {
   /* handle local hot key */
+  if (HasFocus()) {
+    /* Let the InfoBoxManager decide what is to do */
+    InfoBoxManager::OnKeyDown(this, key_code);
```

Comment:
InfoBoxWindows should be agnostic to where they are integrated. This is a layering violation.

Reply:
Sorry, I didn't understand the agnostic word in this context. Can you please explain the violation? Thx.

Reply:
`InfoBoxManager` manages `InfoBoxWindow`, but `InfoBoxWindow` doesn't know about `InfoBoxManager`. And it shouldn't.

Reply:
Thank you, now I understand the violation.
But how to solve this, InfoBoxManager doesn't know about the keys. InfoBoxWindow doesn't know about the other InfoBoxWindow's.
In my opinion the InfoBoxWindow must signal the InfoBoxManager that he should select the next InfoBoxWindow.

Reply:
There used to be code for this, but I removed it 4 years ago: https://github.com/XCSoar/XCSoar/commit/e8c8cde7b1f8566b1e361a4be9b07bba631af40c
Note that the arrow keys are already taken for other actions on InfoBoxes.